### PR TITLE
lint: removes unnecesary use of parens on statements

### PR DESCRIPTION
**Pull request by alfredodeza**
**What this PR does / why we need it**: There is no need for parentheses on statements. For example this:

```python
    return(1)
```

Is the same as:

```python
    return 1
```

Although stylistic, the unnecessary parentheses can cause confusion, making a contributor double-check what is actually occurring (e.g. a tuple or a callable?). 

This PR also adds `make lint` to the CI config so that it prevents newer PRs from getting it in.

**Which issue this PR fixes** Partially fixes issue #https://github.com/anchore/anchore-engine/issues/364

**Special notes**: I *think* I managed to get this right, I double checked, but will need to rely on tests to ensure I didn't remove something that was actually needed

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -88,7 +88,15 @@ jobs:
     steps:
       - checkout
       - run_tests:
-          test_type: unit
+          test_type: test-unit
+
+  lint:
```
Comment (alfredodeza) [inline .circleci/config.yml]: @Btodhunter this didn't seem like it got picked up in the current PR. Is this something that is going to be triggered *after* this is merged? I could swear I've seen configs to be executed in the same PR that implements them

Comment (Btodhunter) [reply .circleci/config.yml]: It will run on the same PR that implements it, you just need to add a lint job to the workflows.

**Comment by zhill** [comment]
To be clear: this isn't needed for 0.7.0 release, correct?

**Comment by alfredodeza** [comment]
It isn't needed for the release, no.

**Comment by alfredodeza** [comment]
Now that the release is almost behind us, would love to get this reviewed and merged, because it touches on so many different places that rebasing is a pain :(

File: .circleci/config.yml [hunk 2]
```diff
@@ -198,6 +206,7 @@ workflows:
   default_workflow:
```
Comment (Btodhunter) [inline .circleci/config.yml]: The `lint` job should gate either unit_tests or integration_tests & build to fail the pipeline early when the linter fails. The lint job should also be added to the RC & Rebuild workflows (a rebase might be needed to see where lint fits into the RC workflow)

Comment (alfredodeza) [reply .circleci/config.yml:206]: I've added linting for the RC, Build, and Rebuild workflows. I don't think we should gate on unit tests or integration tests however. As a developer, I would like to see all unit tests and integration passing even if the linter complains. 

Preventing a build does sound great though, so I've made sure it is required on those workflows.

Comment (Btodhunter) [reply .circleci/config.yml:206]: lgtm!

File: .circleci/config.yml [hunk 3]
```diff
@@ -231,8 +241,12 @@ workflows:
 
   rc_image_workflow:
     jobs:
+      - lint:
+          filters: *filter_rebuild_tags
```
Comment (Btodhunter) [inline .circleci/config.yml:245]: this should be `*filter_rc_tags`

Comment (alfredodeza) [reply .circleci/config.yml:245]: oof good catch

File: .circleci/config.yml [hunk 4]
```diff
@@ -91,7 +91,15 @@ jobs:
     steps:
       - checkout
       - run_tests:
-          test_type: unit
+          test_type: test-unit
+
+  lint:
+    docker:
+      - image: circleci/python:3.6
```
Comment (Btodhunter) [inline .circleci/config.yml:98]: can you add `environment: *global_environment_vars` to this job to be consistent with the rest of the jobs? I don't think this job will actually _need_ any of the current env vars, but the idea of that anchor was to provide a place to set global vars for every job.

**Comment by alfredodeza** [comment]
@Btodhunter from the CircleCi perspective this is good to go right?

File: .circleci/config.yml (hunk 4 above)
Comment (Btodhunter) [reply .circleci/config.yml:98]: @alfredodeza did your force push overwrite this fix?

Comment (alfredodeza) [reply .circleci/config.yml:98]: Not sure... just pushed again to fix this

**Pull request merged**